src/main/java/uk/gov/ons/census/casesvc/service/QuestionnaireLinkedService.java

```diff
@@ -29,49 +28,86 @@ public QuestionnaireLinkedService(
   }
 
   public void processQuestionnaireLinked(ResponseManagementEvent questionnaireLinkedEvent) {
-    UacDTO uac = questionnaireLinkedEvent.getPayload().getUac();
-    String questionnaireId = uac.getQuestionnaireId();
-    UacQidLink uacQidLink = uacService.findByQid(questionnaireId);
+    UacDTO uacDTO = questionnaireLinkedEvent.getPayload().getUac();
+    UacQidLink uacQidLink = uacService.findByQid(uacDTO.getQuestionnaireId());
 
-    checkQidNotLinkedToAnotherCase(uac, uacQidLink);
+    // if the uacqidLink is already linked to a dfiferent case this throws an Exception
+    checkQidNotLinkedToAnotherCase(uacDTO, uacQidLink);
 
-    Case caze;
+    logEvent(questionnaireLinkedEvent, uacQidLink);
 
-    if (isIndividualQuestionnaireType(questionnaireId)) {
-      Case householdCase = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
-      caze = caseService.prepareIndividualResponseCaseFromParentCase(householdCase);
+    Case caze = getCase(uacDTO);
+    uacQidLink.setCaze(caze);
+
+    if (uacQidLink.isBlankQuestionnaireReceived()) {
+      processBlankQuestionnaireLinked(uacQidLink);
+      return;
+    }
 
-      caseService.emitCaseCreatedEvent(caze);
-    } else {
-      caze = caseService.getCaseByCaseId(UUID.fromString(uac.getCaseId()));
+    if (caze.isCcsCase()) {
```

Comment:
ugh. eww. yuk. Do we really have to treat CCS cases so differently?

Reply:
This has isolated the CSS faff for now.  Hopefully we'll get the green light to treat them like normal cases (just emit them) and then all the 'is CCS' stuff can go